src/main/java/model/tenant/User.java

```diff
@@ -36,11 +37,13 @@ public boolean equals(Object o) {
         if (this == o) return true;
         if (o == null || getClass() != o.getClass()) return false;
         User user = (User) o;
-        return Objects.equals(email, user.email);
+
+        return (   Objects.equals(email, user.email)
+                && Objects.equals(userName, user.userName));
     }
 
     @Override
     public int hashCode() {
-        return Objects.hash(email);
+        return Objects.hash(userName);
```

Comment:
I think we should concatenate the username and email hash since the equals also relies on these two at the moment.